File: problems/0881 - Boats to Save People/Go/Solution.go
func numRescueBoats(people []int, limit int) int {
    sort.Ints(people)
    numBoats, start := 0, 0
    
    for end := len(people)-1; start <= end; end-- {
        if end > 0 && people[end] + people[end-1] <= limit {
            end--
        } else if start < end && people[end] + people[start] <= limit {
            start++
        }
        
        numBoats++
    }
    
    return numBoats
}

File: problems/0881 - Boats to Save People/PHP/Solution.php
<?php

class Solution {

    /**
     * @param Integer[] $people
     * @param Integer $limit
     * @return Integer
     */
    function numRescueBoats($people, $limit) {
        sort($people);
        $res = 0;
        $start = 0;
        
        for ($end = count($people)-1; $start <= $end; $end--) {
            if (
                $end > 0 
                && $people[$end] + $people[$end-1] <= $limit
            ) {
                $end--;    
            } elseif (
                $start < $end
                && $people[$end] + $people[$start] <= $limit
            ) {
                $start++;
            }
            
            $res++;
        }
        
        return $res;
    }
}